packages/styles/components/switch.css

```diff
@@ -57,29 +85,92 @@
   .switch[data-selected="true"][data-pressed="true"] & {
     @apply bg-accent-hover;
   }
+
+  @media (prefers-reduced-motion: reduce) {
+    transition: none;
+  }
+}
+
+/* Size-specific control dimensions */
+.switch--sm .switch__control {
+  @apply h-[18px] w-[36px] sm:h-[16px] sm:w-[28px];
+}
+
+.switch--lg .switch__control {
+  @apply h-[32px] w-[52px] sm:h-[24px] sm:w-[40px];
 }
 
 /* Switch thumb */
 .switch__thumb {
-  @apply bg-default-foreground/20 flex size-5 translate-x-0.5 rounded-full shadow-sm sm:size-4;
+  /* Default size (medium) with margin-based positioning */
+  @apply flex size-5 rounded-full bg-white text-black sm:size-4;
   transition:
-    translate 150ms var(--ease-out-quad),
+    margin 150ms var(--ease-out-quad),
     background-color 150ms var(--ease-smooth);
+  margin-inline-start: 2px;
```

Comment:
seems not taking effect

<img width="219" height="78" alt="Image" src="https://github.com/user-attachments/assets/7e03f97c-f636-4295-baa0-c35c8fb692cc" />